tests-ui/tests/store/firebaseAuthStore.test.ts

```diff
@@ -331,6 +331,70 @@ describe('useFirebaseAuthStore', () => {
       const tokenAfterLogout = await store.getIdToken()
       expect(tokenAfterLogout).toBeNull()
     })
+
+    it('should handle network errors gracefully when offline (reproduces issue #4468)', async () => {
+      // This test reproduces the issue where Firebase Auth makes network requests when offline
+      // and fails without graceful error handling, causing toast error messages
+
+      // Simulate a user with an expired token that requires network refresh
+      mockUser.getIdToken.mockReset()
+
+      // Mock network failure (auth/network-request-failed error from Firebase)
+      const networkError = new Error(
+        'Firebase: Error (auth/network-request-failed).'
+      )
+      networkError.name = 'FirebaseError'
+      ;(networkError as any).code = 'auth/network-request-failed'
```

Comment:
Is there _any_ way to avoid `as any`?

Reply:
What is the concrete Error type when it actually runs?

Reply:
this was vibe-coded... guilty as charged. I'll take a look.